README.markdown

```diff
@@ -125,170 +70,157 @@ You can also use this parameter to set a specific registry value as the default
 
-A description of the service.
+#####`path`
 
-#####`command`
+*Optional.* Specifies a Registry value for Puppet to manage. Valid options: a string containing a Registry path. If any of the parent keys in the path don't exist, Puppet creates them automatically. For example: 'HKLM\Software' or 'HKEY_LOCAL_MACHINE\Software\Vendor'. Default value: the title of your declared resource.
 
-Specifies the command to execute.
+If Puppet is running on a 64-bit system, the 32-bit Registry key can be explicitly managed using a prefix. For example: '32:HKLM\Software\Value3'.
 
-#####`start`
+#####`ensure`
 
-Specifies the starting mode of the service. Valid values are 'automatic', 'manual', and 'disabled'. 
+*Optional.* Tells Puppet whether the value should or shouldn't exist. Valid options: 'present' and 'absent'. Default value: unspecified (if left unspecified, Puppet won't enforce any other parameters unless the value is already present on a node.)
 
-The [native service resource](http://docs.puppetlabs.com/references/latest/type.html#service) can also be used to manage this setting.
+#####`type`
 
-####Extended `registry::service` example
+*Optional.* Sets the data type of the specified value. Valid options: 'string', 'array', 'dword', 'qword', 'binary', and 'expand'. Default value: 'string'.
 
-    class registry::service_example {
-      # Define a new service named "Puppet Test" that is disabled.
-      registry::service { 'PuppetExample1':
-        display_name => "Puppet Example 1",
-        description  => "This is a simple example managing the registry entries for a Windows Service",
-        command      => 'C:\PuppetExample1.bat',
-        start        => 'disabled',
-      }
-      registry::service { 'PuppetExample2':
-        display_name => "Puppet Example 2",
-        description  => "This is a simple example managing the registry entries for a Windows Service",
-        command      => 'C:\PuppetExample2.bat',
-        start        => 'disabled',
-      }
-    }
+#####`data`
 
-###Purge Values
+*Required.* Provides the contents of the specified value. Valid options: a string by default; an array if specified through the `type` parameter.
 
-The Registry module provides a means of ensuring that only Puppet-specified values are associated with a particular key. This is not default behavior, but a feature you must enable.
+####`registry::service`
 
-In order to make sure only the values specified via Puppet are associated
-with a particular key, use the `purge_values => true` parameter of the
-`registry_key` resource. **Enabling this feature will delete any values not managed by Puppet.** 
+#####`ensure`
 
-The `registry::purge_example` class provides a quick and easy way to see a demonstration of how this works. This example class has two modes of operation determined by the Facter fact `PURGE_EXAMPLE_MODE`: 'setup' or 'purge'.
+*Required.* Tells Puppet whether the service should or shouldn't exist. Valid options: 'present' and 'absent'.
```

Comment:
Not required, default is unset which gets changed to 'present' if not specified.